modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultIRTreeToASMBytesPhase.java

```diff
@@ -0,0 +1,1655 @@
+        methodWriter.writeStatementOffset(irDeclarationNode.getLocation());
+
+        Variable variable = writeScope.defineVariable(irDeclarationNode.getDeclarationType(), irDeclarationNode.getName());
+
+        if (irDeclarationNode.getExpressionNode() ==  null) {
+            Class<?> sort = variable.getType();
+
+            if (sort == void.class || sort == boolean.class || sort == byte.class ||
+                    sort == short.class || sort == char.class || sort == int.class) {
+                methodWriter.push(0);
+            } else if (sort == long.class) {
+                methodWriter.push(0L);
+            } else if (sort == float.class) {
+                methodWriter.push(0F);
+            } else if (sort == double.class) {
+                methodWriter.push(0D);
+            } else {
+                methodWriter.visitInsn(Opcodes.ACONST_NULL);
+            }
+        } else {
+            visit(irDeclarationNode.getExpressionNode(), writeScope);
+        }
+
+        methodWriter.visitVarInsn(variable.getAsmType().getOpcode(Opcodes.ISTORE), variable.getSlot());
+    }
+
+    @Override
+    public void visitReturn(ReturnNode irReturnNode, WriteScope writeScope) {
+        MethodWriter methodWriter = writeScope.getMethodWriter();
+        methodWriter.writeStatementOffset(irReturnNode.getLocation());
+
+        if (irReturnNode.getExpressionNode() != null) {
+            visit(irReturnNode.getExpressionNode(), writeScope);
+        }
+
+        methodWriter.returnValue();
+    }
+
+    @Override
+    public void visitStatementExpression(StatementExpressionNode irStatementExpressionNode, WriteScope writeScope) {
+        MethodWriter methodWriter = writeScope.getMethodWriter();
+        methodWriter.writeStatementOffset(irStatementExpressionNode.getLocation());
+        visit(irStatementExpressionNode.getExpressionNode(), writeScope);
+        methodWriter.writePop(MethodWriter.getType(irStatementExpressionNode.getExpressionNode().getExpressionType()).getSize());
+    }
+
+    @Override
+    public void visitTry(TryNode irTryNode, WriteScope writeScope) {
+        MethodWriter methodWriter = writeScope.getMethodWriter();
+        methodWriter.writeStatementOffset(irTryNode.getLocation());
+
+        Label tryBeginLabel = new Label();
+        Label tryEndLabel = new Label();
+        Label catchesEndLabel = new Label();
+
+        methodWriter.mark(tryBeginLabel);
+
+        visit(irTryNode.getBlockNode(), writeScope.newBlockScope());
+
+        if (irTryNode.getBlockNode().doAllEscape() == false) {
+            methodWriter.goTo(catchesEndLabel);
+        }
+
+        methodWriter.mark(tryEndLabel);
+
+        List<CatchNode> catchNodes = irTryNode.getCatchNodes();
+
+        for (int i = 0; i < catchNodes.size(); ++i) {
+            CatchNode irCatchNode = catchNodes.get(i);
+            Label catchJumpLabel = catchNodes.size() > 1 && i < catchNodes.size() - 1 ? catchesEndLabel : null;
```

Comment:
put some parens around this condition.

Reply:
Pulled this into a boolean to make it clearer.